google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -137,22 +169,119 @@ public static Timestamp parseTimestamp(String timestamp) {
     }
     String secondsPart = matcher.group(1);
     String nanosPart = matcher.group(2);
-    long seconds;
-    seconds = format.parseMillis(secondsPart) / 1000;
+    long seconds = TimeUnit.MILLISECONDS.toSeconds(ISO_FORMAT_UTC.parseMillis(secondsPart));
     int nanos = 0;
     if (nanosPart != null) {
-      String padded = Strings.padEnd(nanosPart.substring(1), 9, '0');
-      nanos = Integer.parseInt(padded);
+      nanos = parseFractToNanos(nanosPart.substring(1));
+    }
+    return ofTimeSecondsAndNanos(seconds, nanos);
+  }
+
+  /**
+   * Creates a Timestamp instance from the given string. String is in the format
+   * yyyy-MM-dd HH:mm:ss.SSSSSSZZ. If no timezone is given, UTC is assumed.
+   */
+  public static Timestamp parseCloudTimestamp(String timestamp) {
```

Comment:
Ahh I missed this response. The problem is that there are two possible things before the '.':

1. Date only
2. Date and time

I would have to use two separate parsers like I am already doing.